searchcore/src/vespa/searchcore/proton/matching/match_tools.cpp

```diff
@@ -214,8 +214,9 @@ MatchToolsFactory::createDiversifier() const
         LOG(warning, "Skipping diversity due to no %s attribute.", _diversityParams.attribute.c_str());
         return std::unique_ptr<IDiversifier>();
     }
-    size_t max_per_group = _rankSetup.getHeapSize()/_diversityParams.min_groups;
-    return DiversityFilter::create(*attr, _rankSetup.getHeapSize(), max_per_group, _diversityParams.min_groups,
+    uint32_t heapSize = hitcollector::HeapSize::lookup(_queryEnv.getProperties(), _rankSetup.getHeapSize());
```

Comment:
Do not look it up again here. Just take heapSize in as an argument to createDiversity(uint32 heapSize);
It has already been resolved at the callsite and is present as params.heapSize in MatchMaster::match